packages/runtime-dom/types/jsx.d.ts

```diff
@@ -1331,6 +1331,32 @@ type NativeElements = {
   >
 }
 
+type AnyEmits = { [evtName: string]: any; };
+
+/** get jsx attr props object from DefineComponent */
+type ExtractEmitProps<C extends Record<string, any>> = (
+  C['emits'] extends (
+    (infer EmitObj & ThisType<void>) |
+    (Array<infer EvtNames> & ThisType<void>) |
+    undefined
+  )
+    ? Exclude<EmitObj, string[]> extends never
+      ? (
+        // @ts-ignore TODO ts extends covariance
+        { [K in `on${Capitalize<EvtNames>}`]?: (payload?: any) => any; }
+      )
+      : (
+        AnyEmits extends EmitObj
+          ? {}
+          : GetEmitPropsFromEmitOption<Exclude<EmitObj, string[]>>
+      )
+    : {}
+);
+
+type GetEmitPropsFromEmitOption<T extends Record<string, any>> = {
+  [K in keyof T as `on${Capitalize<string &K>}`]?: (...payload: Parameters<T[K]>) => any
+};
+
```

Comment:
Remove this, and add the following code to `componentEmits.ts`, WDYT?:
```ts
export type ExtractEmitProps<Options extends EmitsOptions> = Options extends (Array<infer EvtNames>)
  ? {
      [key in `on${Capitalize<string &EvtNames>}`]: (...args: any[]) => any
    }
  : Exclude<Options, ThisType<void>> extends ObjectEmitsOptions
    ? {
        [key in keyof Options as `on${Capitalize<string &key>}`]?: Options[key] extends ((...args: infer Args) => infer R)
          ? (...args: Args) => R
          : (...args: any[]) => any
      }
    : {}
```

Reply:
With `Capitalize`:
`emit('modelValue') -> onModel-value` 

need transform to camelcase